src/main/java/org/zalando/nakadi/filters/LoggingFilter.java

```diff
@@ -134,35 +122,75 @@ protected void doFilterInternal(final HttpServletRequest request,
-            final Long timing = currentTime - requestLogInfo.requestTime;
-
-            ACCESS_LOGGER.info("{} \"{}{}\" \"{}\" \"{}\" {} {}ms \"{}\" \"{}\" {}B",
-                    requestLogInfo.method,
-                    requestLogInfo.path,
-                    requestLogInfo.query,
-                    requestLogInfo.userAgent,
-                    requestLogInfo.user,
-                    response.getStatus(),
-                    timing,
-                    requestLogInfo.contentEncoding,
-                    requestLogInfo.acceptEncoding,
-                    requestLogInfo.contentLength);
-            nakadiKpiPublisher.publish(accessLogEventType, () -> new JSONObject()
-                    .put("method", requestLogInfo.method)
-                    .put("path", requestLogInfo.path)
-                    .put("query", requestLogInfo.query)
-                    .put("app", requestLogInfo.user)
-                    .put("app_hashed", nakadiKpiPublisher.hash(requestLogInfo.user))
-                    .put("status_code", response.getStatus())
-                    .put("response_time_ms", timing));
+    private void logRequest(final RequestLogInfo requestLogInfo, final int statusCode) {
+        final Long timeSpentMs = System.currentTimeMillis() - requestLogInfo.requestTime;
+
+        logToAccessLog(requestLogInfo, statusCode, timeSpentMs);
+        logToNakadi(requestLogInfo, statusCode, timeSpentMs);
+        traceRequest(requestLogInfo, statusCode, timeSpentMs);
+    }
+
+    private void logToNakadi(final RequestLogInfo requestLogInfo, final int statusCode, final Long timeSpentMs) {
+        nakadiKpiPublisher.publish(accessLogEventType, () -> new JSONObject()
+                .put("method", requestLogInfo.method)
+                .put("path", requestLogInfo.path)
+                .put("query", requestLogInfo.query)
+                .put("app", requestLogInfo.user)
+                .put("app_hashed", nakadiKpiPublisher.hash(requestLogInfo.user))
+                .put("status_code", statusCode)
+                .put("response_time_ms", timeSpentMs));
+    }
+
+    private void logToAccessLog(final RequestLogInfo requestLogInfo, final int statusCode, final Long timeSpentMs) {
+        ACCESS_LOGGER.info("{} \"{}{}\" \"{}\" \"{}\" {} {}ms \"{}\" \"{}\" {}B",
+                requestLogInfo.method,
+                requestLogInfo.path,
+                requestLogInfo.query,
+                requestLogInfo.userAgent,
+                requestLogInfo.user,
+                statusCode,
+                timeSpentMs,
+                requestLogInfo.contentEncoding,
+                requestLogInfo.acceptEncoding,
+                requestLogInfo.contentLength);
+    }
+
+    private void traceRequest(final RequestLogInfo requestLogInfo, final int statusCode, final Long timeSpentMs) {
+        if ("POST".equals(requestLogInfo.method) && requestLogInfo.path.startsWith("/event-types/") &&
```

Comment:
Is `path` guaranteed not to be null?

Reply:
I can't guarantee that, so it makes sense to add a check: https://github.com/zalando/nakadi/pull/1027/commits/dbae4615dc693b826bf4d8605c58161cadf62e51